src/sources/internal_metrics.rs

```diff
@@ -17,6 +17,19 @@ pub struct InternalMetricsConfig {
     #[derivative(Default(value = "2"))]
     scrape_interval_secs: u64,
     tags: TagsConfig,
+    namespace: Option<String>,
+}
+
+impl InternalMetricsConfig {
+    // TODO: Remove this annotation once it's actually being used.
+    #[allow(dead_code)]
+    /// Override the default namespace.
+    fn namespace<T: Into<String>>(namespace: T) -> Self {
+        Self {
+            namespace: Some(namespace.into()),
+            ..Self::default()
+        }
+    }
```

Comment:
Nit: I might have left this out until you were actually using it.

Reply:
Thanks, I almost did, it just didn't feel like a complete API without it!

It'll be used in a follow-up PR so this TODO will be moot shortly.